unit/java_bytecode/java_bytecode_parse_generics/GenericBounds.java

```diff
@@ -0,0 +1,15 @@
+class GenericContainer<T> {
+}
+
+class GenericBounds extends GenericContainer<Class<?>> {
+    // references exist only to load these class files, too
+    GenericBoundsUpper upper;
+    GenericBoundsLower lower;
+
+}
+
+class GenericBoundsUpper extends GenericContainer<Class<? extends Class>> {
+}
+
+class GenericBoundsLower extends GenericContainer<Class<? super Class>> {
+}
```

Comment:
We should also have examples of classes that implement interfaces with unsupported signatures.

Reply:
I added a generic interface test and moved the source to `DerivedGenerics.java`